dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/relationship/HibernateRelationshipStore.java

```diff
@@ -317,7 +317,8 @@ private <T extends SoftDeletableObject> Predicate[] whereConditionPredicates(
 
     toSubQuery.where(
         builder.equal(root.get("to"), toRoot.get("id")),
-        builder.equal(toRoot.get(relationshipEntityType), entity.getId()));
+        builder.equal(toRoot.get(relationshipEntityType), entity.getId()),
+        builder.equal(root.get("relationshipType").get("bidirectional"), true));
```

Comment:
why is this needed?

Reply:
Because we need to implement the `rule` also in the store.
In `tracker/relationships?trackedEntity={TEa}` we need to return only the relationships originated from `TEa`, so we need to include in the query the relationships where `TEa` is on the `to` side only if the relationship is bidirectional.

In a perfect world I would like to have this rule written once in the code and used by the different place that need to apply it, but I think we will not be able to get away from the duplication at least in the service and in the store.